setup.py

```diff
@@ -39,7 +39,7 @@
/nitpick/)\n[![Project License](https://img.shields.io/pypi/l/nitpick.svg)](https://pypi.org/project/nitpick/)\n[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)\n[![Dependabot Status](https://api.dependabot.com/badges/status?host=github&repo=andreoliwa/nitpick)](https://dependabot.com)\n[![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)\n\nFlake8 plugin to enforce the same tool configuration ([flake8](https://gitlab.com/pycqa/flake8), [isort](https://github.com/timothycrosley/isort), [mypy](http://mypy-lang.org/), [Pylint](https://www.pylint.org)...) across multiple Python projects.\n\nUseful if you maintain multiple projects and want to use the same configs in all of them.\n\n## Style file\n\nA "nitpick code style" is a [TOML](https://github.com/toml-lang/toml) file with the settings that should be present in config files from other tools.\n\nExample of a style:\n\n```\n["pyproject.toml".tool.black]\nline-length = 120\n\n["pyproject.toml".tool.poetry.dev-dependencies]\npylint = "*"\n\n["setup.cfg".flake8]\nignore = "D107,D202,D203,D401"\nmax-line-length = 120\ninline-quotes = "double"\n\n["setup.cfg".isort]\nline_length = 120\nmulti_line_output = 3\ninclude_trailing_comma = true\nforce_grid_wrap = 0\ncombine_as_imports = true\n```\n\nThis style will assert that:\n\n- ... [black](https://github.com/psf/black), [isort](https://github.com/timothycrosley/isort) and [flake8](https://gitlab.com/pycqa/flake8) have a line length of 120;\n- ... [flake8](https://gitlab.com/pycqa/flake8) and [isort](https://github.com/timothycrosley/isort) are configured as above in `setup.cfg`;\n- ... [Pylint](https://www.pylint.org) is present as a [Poetry](https://github.com/python-poetry/poetry) dev dependency in `pyproject.toml`).\n\n## Quick setup\n\nTo try the package, simply install it (in a virtualenv or globally, wherever) and run `flake8`:\n\n    $ pip install -U nitpick\n    $ flake8\n\nNitpick will download and use the opinionated [default style file](https://raw.githubusercontent.com/andreoliwa/nitpick/v0.21.4/nitpick-style.toml).\n\nYou can use it as a template to configure your own style.\n\n### Run as a pre-commit hook (recommended)\n\nIf you use [pre-commit](https://pre-commit.com/) on your project (you should), add this to the `.pre-commit-config.yaml` in your repository:\n\n    repos:\n      - repo: https://github.com/andreoliwa/nitpick\n        rev: v0.21.4\n        hooks:\n          - id: nitpick\n\nTo install the `pre-commit` and `commit-msg` Git hooks:\n\n    pre-commit install --install-hooks\n    pre-commit install -t commit-msg\n\nTo start checking all your code against the default rules:\n\n    pre-commit run --all-files\n\n---\n\nFor more details on styles and which configuration files are currently supported, [see the full documentation](https://nitpick.rtfd.io/).\n',
```

Comment:
Are you sure you need both `setup.py` and `poetry`? 
I am using only `poetry` for several year now and it works just fine.

Reply:
Yeah, I will see if I can remove it now.
It was a dirty hack for older Poetry versions.

Are you able to use `pip install -e` or some similar command?
I remember the need for `setup.py` was related to it.

Reply:
You don't need `pip install -e` with `poetry`.
Because it installs your package in editable mode by default.

Reply:
I was reading [my own comments](https://github.com/andreoliwa/nitpick/blob/eda7d25e11425c1a9eeababe8f9d3e8b52f24f80/Makefile#L61-L64), and I found out what was the reason for this `setup.py`:

```shell
23:52 $ pip install -e ~/Code/nitpick/
ERROR: File "setup.py" not found. Directory cannot be installed in editable mode: /Users/aa/Code/nitpick
(A "pyproject.toml" file was found, but editable mode currently requires a setup.py based build.)
```

Without a `setup.py`, I could not install an editable version of Nitpick from a local directory with `pip`.
I was modifying Nitpick on my laptop (without pushing to GitHub), and testing the modifications on another local project.

Do you know a better way to do that (use Nitpick on another project, both on the laptop)?

Reply:
What do you want to achieve exactly?

`poetry install` will make your project editable in the target virtual environment.
Maybe you want to change which target `poetry` is using?

Reply:
Is it not what [`poetry add <LOCAL_PATH_TO_PROJECT>`](https://python-poetry.org/docs/cli/#add) is for?

> Path dependencies pointing to a local directory will be installed in editable mode (i.e. setuptools "develop mode"). It means that changes in the local directory will be reflected directly in environment.

Reply:
I did this a long time ago, Poetry version was around 0.11 or 0.12.
Either `poetry add ../relative/path` was not working back then or I was doing it wrong.

I will remove the `setup.py`.
Thanks guys for the helpful tips. 😉 👍